src/frontend/src/tests/lib/services/rest.services.spec.ts

```diff
@@ -0,0 +1,167 @@
+import { restRequest } from '$lib/services/rest.services';
+import { expect } from 'vitest';
+
+describe('rest.services', () => {
+	describe('restRequest', () => {
+		const mockSuccessfulRequest = vi.fn().mockResolvedValue('success');
+		const mockFailedRequest = vi.fn().mockRejectedValue(new Error('Failed'));
+
+		const mockOnSuccess = vi.fn();
+		const mockOnError = vi.fn();
+		const mockOnRetry = vi.fn();
+
+		// we mock console.error and console.warn just to avoid unnecessary logs while running the tests
+		vi.spyOn(console, 'error').mockImplementation(() => undefined);
+		vi.spyOn(console, 'warn').mockImplementation(() => undefined);
+
+		beforeEach(() => {
+			vi.clearAllMocks();
+		});
+
+		it('should call onSuccess when the request succeeds on the first try', async () => {
```

Comment:
If I'm not mistaken, we need another test to assert the call succeed after first but before max.

Reply:
i created one that is `should succeed after retries if a later attempt is successful`, is that ok? or something else?

Reply:
all good